res/controllers/Numark-Mixtrack-3-scripts.js

```diff
@@ -1819,119 +1547,95 @@ NumarkMixtrack3.WheelMove = function(channel, control, value, status, group) {
-        } else {
-            engine.setValue("[Channel" + decknum + "]", "hotcue_" + hotCue + "_activate", 0);
-        }
+        engine.setValue(deck.group, "hotcue_" + hotCue + "_activate", value === DOWN);
     }
 };
 
 NumarkMixtrack3.OnHotcueChange = function(value, group, control) {
-    var decknum = script.deckFromGroup(group);
-    var deck = NumarkMixtrack3.decks["D" + decknum];
+    var deck = NumarkMixtrack3.deckFromGroup(group);
     var padindex = parseInt(NumarkMixtrack3.channelRegEx.exec(control)[1]);
 
     deck.LEDs["hotCue" + padindex].onOff((value) ? ON : OFF);
 };
 
-NumarkMixtrack3.SamplerButton = function(channel, control, value, status,
-    group) {
+NumarkMixtrack3.SamplerButton = function(channel, control, value, status, group) {
     var isplaying = engine.getValue(group, "play");
-    var padindex = parseInt(group.substring(8, 9));
-    var decknum = padindex;
-    
-    if (decknum > 4) {
+    var isLoaded = engine.getValue(group, "track_loaded");
+    var padIndex = parseInt(group.substring(8, 9));
+    var sampler = NumarkMixtrack3.samplers["S" + padIndex];
+    var decknum = 1;
+
+    if (padIndex > 4) {
         decknum = 2;
         engine.setValue("[Deere]","sampler_bank_2", true);
     } else {
-        decknum = 1;
         engine.setValue("[Deere]","sampler_bank_1", true);
     }
 
-    decknum = NumarkMixtrack3.deckFromGroup("[Channel" + decknum + "]");
-    var deck = NumarkMixtrack3.decks["D" + decknum];
-    var sampler = NumarkMixtrack3.samplers["S" + padindex];
+    var deck = NumarkMixtrack3.deckFromGroup("[Channel" + decknum + "]");
 
     if (value === DOWN) {
-
-
-    sampler.PADSampleButtonHold.ButtonDown(channel, control, value, status, group);
+        sampler.PADSampleButtonHold.ButtonDown(channel, control, value, status, group);
 
         if (!isplaying) {
             if (deck.shiftKey) {
-                //Shift is on, play sampler with no Sync
+                // shift is on, play sampler with no Sync
                 engine.setValue(group, "beatsync", 0);
                 engine.setValue(group, "cue_gotoandplay", 1);
-                sampler.LEDs["PADsampler" + padindex].flashOn(300,
-                    PADcolors.purple, 300);
+            } else if (!isLoaded) {
+                engine.setValue(group, "LoadSelectedTrack", 1);
+            } else if (deck.TapDown) {
+                engine.setValue(group, "eject", 1);
+                engine.setValue(group, "eject", 0);
             } else {
-                //play sampler with Sync
+                // play sampler with Sync
                 engine.setValue(group, "cue_gotoandplay", 1);
                 engine.setValue(group, "beatsync", 1);
-                sampler.LEDs["PADsampler" + padindex].flashOn(300,
-                    PADcolors.purple, 300);
             }
+
+            sampler.LEDs["PADsampler" + padIndex].flashOn(300, PADcolors.purple, 300);
         } else {
             engine.setValue(group, "stop", 1);
```

Comment:
The pads do turn different colors but I think they are limited to 3 which are used for manual, auto, and sampler modes so that might be confusing. I agree with your suggested change, but perhaps I can have the pad flash slowly while playing.